File: 0136.Single_Number/solution.c
#include <stdio.h>

int singleNumber(int* nums, int numsSize) {
    int ret_val = 0;
    for (int i = 0; i < numsSize; i++)
        ret_val ^= nums[i];
    return ret_val;
}

int main() {
    int nums[] = {4,1,2,1,2};
    int numsSize = sizeof(nums)/sizeof(nums[0]);
    printf("singleNumber=%d\n", singleNumber(nums, numsSize));
    return 0;
}

File: 0136.Single_Number/solution.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        hash_table = {}
        for num in nums:
            if num in hash_table.keys():
                del hash_table[num]
            else:
                hash_table[num] = 1
        return list(hash_table.keys())[0]